File: edukate-common/src/main/java/io/github/sanyavertolet/edukate/common/utils/AuthUtils.java
package io.github.sanyavertolet.edukate.common.utils;

import io.github.sanyavertolet.edukate.common.users.EdukateUserDetails;
import lombok.experimental.UtilityClass;
import org.springframework.security.core.Authentication;
import reactor.core.publisher.Mono;

@UtilityClass
public class AuthUtils {

    public static String id(Authentication authentication) {
        if (authentication == null) {
            return null;
        }
        EdukateUserDetails userDetails = (EdukateUserDetails) authentication.getPrincipal();
        return userDetails.getId();
    }

    public static Mono<String> monoId(Authentication authentication) {
        return Mono.justOrEmpty(id(authentication));
    }
}

File: edukate-common/src/main/kotlin/io/github/sanyavertolet/edukate/common/utils/AuthUtils.kt
package io.github.sanyavertolet.edukate.common.utils

import io.github.sanyavertolet.edukate.common.users.EdukateUserDetails
import org.springframework.security.core.Authentication
import reactor.kotlin.core.publisher.toMono

fun Authentication?.id(): Long? = this?.let { it.principal as EdukateUserDetails }?.id

fun Authentication?.monoId() = id().toMono()
